src/main/java/io/storj/gerrit/plugins/codeowners/ReviewAssigner.java

```diff
@@ -51,21 +54,46 @@ public class ReviewAssigner implements WorkInProgressStateChangedListener, Comme
     // use a global cache to reduce calls to the gerrit APIs
     // [user/email:]name -> gerritAccountId
     private final ConcurrentMap<String, Integer> cache = new ConcurrentHashMap<>();
-    private final Function<String, Integer> loader;
+    private final Function<String, Integer> loadAccountID;
+    private final String reviewerGroup;
 
     @Inject
-    public ReviewAssigner(final GitHub github, final GerritApi gerrit, final GitRepositoryManager git) {
+    public ReviewAssigner(PluginConfigFactory cfg, final GitHub github, final GerritApi gerrit, final GitRepositoryManager git) {
         this.github = github;
         this.gerrit = gerrit;
         this.git = git;
 
-        this.loader = (k) -> {
+        PluginConfig config = cfg.getFromGerritConfig("codeowners");
+        reviewerGroup = config.getString("reviewerGroup", "");
+
+        this.loadAccountID = (String query) -> {
             try {
-                final List<AccountInfo> accounts = this.gerrit.accounts().query(k).get();
+                final List<AccountInfo> candidates = this.gerrit.accounts().query(query).get();
+                if (candidates.size() == 0) {
+                    // did not find a suitable match
+                    return null;
+                }
+
+                AccountInfo candidate = candidates.get(0);
+                if(reviewerGroup == "") {
+                    // no filtering by the reviewer group.
+                    return candidate._accountId;
+                }
 
-                if (accounts.size() > 0) {
-                    return accounts.get(0)._accountId;
+                final List<GroupInfo> groups = this.gerrit.groups().query(reviewerGroup).get();
+                if(groups.size() == 0) {
+                    log.error("failed to find reviewerGroup");
+                    return null;
                 }
+
+                for(GroupInfo group: groups) {
+                    if(group.members.contains(candidate)) {
```

Comment:
Don't we need a loop to check all the candidates? As far as I understand the `candidate` here is always `candidates.get(0)`, and if she or he doesn't part the group we don't check the next candidate...

Reply:
Oh, I see. I was misled by the `s/account/candidates/g` rename here. These are not candidates, this part is just retrieving the account information.

Candidates are handled below in `assign`. 

For me, the original `accounts` variable name was more expressive ;-)

Reply:
Sure, I'll change it back.